tools/tpm2_quote.c

```diff
@@ -205,6 +210,80 @@ static UINT16  calcSizeofTPMT_SIGNATURE( TPMT_SIGNATURE *sig )
     return size > sizeof(*sig) ? sizeof(*sig) : size;
 }
 
+static bool write_tss_file(TPM2B_ATTEST *quoted, TPMT_SIGNATURE *signature,
+        const char *path) {
+    bool res = false;
+    size_t num_bytes;
+    FILE *fp = fopen(path,"wb");
+
+    if (!fp) {
+        printf("Output file %s cannot be created: %s\n",
+                path, strerror(errno));
+        goto error;
+    }
+
+    num_bytes = calcSizeofTPM2B_ATTEST(quoted);
```

Comment:
Where did this macro come from sapi headers?

Couldn't one use:
```
BUFFER_SIZE(typeof(quoted), field)
```

Reply:
I really don't want to touch this, because the `calcSizeofTPM2B_ATTEST()` uses the size field of the buffer while `BUFFER_SIZE()` uses `sizeof` the structure.

In my tests the data returned by Tss2_Sys_Quote() is much less than the size of the TPM2B_ATTEST structure for some reason. Either I don't understand some things here, or this might be worth opening an issue. However, changing to `BUFFER_SIZE()` would cause hundreds of zero bytes to be output into the quote file - at least on my machine.

The TPM chip actually only returns 121 bytes of data, while the structure is 1280 bytes large.

Reply:
The statement, "I really don't want to touch this, because the calcSizeofTPM2B_ATTEST() uses the size field of the buffer while BUFFER_SIZE() uses sizeof the structure." is 100% wrong,

`BUFFER_SIZE()` uses the field within the struct.

Reply:
I see your using the size stored in the structure, but why are you adding 2?

Reply:
Ahh I see, now it makes sense it's not in the diffstat. Thos calcSize routines were inherited and already apart of tpm2_quote. Those routines look very wrong. I would remove those, as that's the cruft we want to get rid of in the repository.

We should be using the size fields and sizeof() and other non-magic ways to determine the sizes of buffers, or just serialize the buffer to disk.

The current problem of casting a struct to a byte() array, is that any padding goes to disk with it, and the file becomes machine dependent.